compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -592,14 +592,13 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
                 &cause,
                 hir.get_if_local(impl_m.def_id)
                     .and_then(|node| node.fn_decl())
-                    .map(|decl| (decl.output.span(), Cow::from("return type in trait"))),
+                    .map(|decl| (decl.output.span(), Cow::from("return type in trait"), false)),
```

Comment:
Question (not-blocking): do you think it's worth adding a simple enum just so it's more obvious what is parameter is? Reading this at face value it's very not obvious:

```rs
enum NoteOrder {
    PrimaryFirst,
    SecondaryFirst
}
```

the callsite will then read like

```rs
.map(|decl| (
    decl.output.span(),
    Cow::from("return type in trait"), 
    NoteOrder::PrimaryFirst
)),
```